src/main/java/de/uni_passau/fim/se2/pipeline_helper/checkers/file_exists/FileExistsChecker.java

```diff
@@ -0,0 +1,95 @@
+// SPDX-FileCopyrightText: 2022 Pipeline Helper Contributors
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+package de.uni_passau.fim.se2.pipeline_helper.checkers.file_exists;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import de.uni_passau.fim.se2.pipeline_helper.model.Checker;
+import de.uni_passau.fim.se2.pipeline_helper.model.CheckerException;
+import de.uni_passau.fim.se2.pipeline_helper.model.CheckerResult;
+
+/**
+ * Checks that all given paths are readable files and are not empty.
+ */
+public class FileExistsChecker implements Checker {
+
+    private static final String CHECKER_NAME = "FileExistsChecker";
+
+    private final List<Path> toCheck;
+
+    public FileExistsChecker(final List<Path> toCheck) {
+        this.toCheck = toCheck;
+    }
+
+    @Override
+    public CheckerResult check() throws CheckerException {
+        final Map<FileStatus, List<Path>> filesByStatus = new HashMap<>();
+        Arrays.stream(FileStatus.values()).forEach(status -> filesByStatus.put(status, new ArrayList<>()));
+
+        for (final Path p : toCheck) {
+            if (!Files.exists(p)) {
+                filesByStatus.get(FileStatus.MISSING).add(p);
+                continue;
+            }
+
+            try {
+                if (Files.isRegularFile(p) && isFileEmpty(p)) {
```

Comment:
Creating a directory instead of a file the tool looks for produces a successful result:

```bash
mkdir Tests.txt
java -jar ./target/pipeline-helper-2.2.1-SNAPSHOT-shaded.jar file-exists Tests.txt
cat customFeedbacks/TEST-FileExistsChecker.json
```
shows
```json
{"name":"FileExistsChecker","successful":true}
```